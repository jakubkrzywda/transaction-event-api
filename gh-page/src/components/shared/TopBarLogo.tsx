import React from "react";
import logo from "../../img/logo.svg";

const TopBarLogo = () => {
    return <div className="logo"><img src={logo} alt="Logo" height="90" /></div>;
};

export default TopBarLogo;