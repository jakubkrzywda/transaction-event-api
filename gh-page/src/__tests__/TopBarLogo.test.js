import React from "react";
import Enzyme, {shallow} from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import TopBarLogo from "../components/shared/TopBarLogo";
import logo from "../img/logo.svg";

Enzyme.configure({adapter: new Adapter()});

describe("TopBarLogo", () => {
    it("renders without crashing", () => {
        const wrapper = shallow(<TopBarLogo />);
        expect(wrapper.exists()).toBe(true);
    });
  
    it("matches snapshot", () => {
        const wrapper = shallow(<TopBarLogo />);
        expect(wrapper).toMatchSnapshot();
    });
    
    it("base classname matches", () => {
        const wrapper = shallow(<TopBarLogo />);
        expect(wrapper.hasClass("logo")).toBe(true);
    });
      
    it("renders img", () => {
        const wrapper = shallow(<TopBarLogo />);
        expect(wrapper.find("img")).toBeDefined();
    });
    
    it("renders img with src", () => {
        const wrapper = shallow(<TopBarLogo />);
        expect(wrapper.find("img").prop("src")).toBe(logo);
    });
    
    it("renders img with alt", () => {
        const wrapper = shallow(<TopBarLogo />);
        expect(wrapper.find("img").prop("alt")).toBe("Logo");
    });
    
    it("renders img with height", () => {
        const wrapper = shallow(<TopBarLogo />);
        expect(wrapper.find("img").prop("height")).toBe("90");
    });
});