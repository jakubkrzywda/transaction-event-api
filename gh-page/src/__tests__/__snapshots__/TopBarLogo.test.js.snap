// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopBarLogo matches snapshot 1`] = `ShallowWrapper {}`;
